src/vhip/mlmodel/compute_ml_features.py

```diff
@@ -459,6 +476,35 @@ def compute_feature(self, pair: Pairs) -> Pairs:
 
         pair.GCdifference = self.GCcontent[pair.virus] - self.GCcontent[pair.host]  # pyright: ignore
 
+        # Create CodonBiasComparison objects for the pair
+        virus = pair.virus.replace(
+            self.genome_ext, self.gene_ext
+        )  # first get the correct key based on genes file name (virus)
```

Comment:
can you explain how the code works there?

Reply:
This is to reconcile the fact that in the pre-existing code, key values for dictionaries are genome filenames (aka, organism name + genome extension). Pairs in determine_pairs form attributes .virus and .host by whatever string you input, which has historically been the genome file name. I want to change this so that Pairs attributes .virus and .host are organism names only (with no file extension), but this is not so simple. All of the genome feature calculations use genome filenames as dictionary keys, and all of the gene feature calculations use gene filenames as dictionary keys. It's likely that several tests would need to be updated too. I can do all this - but I think it would justify it's own PR... ^These couple of lines of code satisfy the discrepancy for now: pair.virus and pair.host are used to access values for the genome features (above this code), and virus and host (which are here defined as just pair.virus and pair.host with the file extension switched to genes) are used to access values for the gene features. Let me know what you think about proceeding!